ravendb/rvn_server.go

```diff
@@ -0,0 +1,760 @@
+		if err == nil {
+			break
+		} else {
+			if i != numberOfIterationsToDial {
+				time.Sleep(time.Second * 5)
+			} else {
+				return err
+			}
+		}
+	}
+
+	defer conn.Close()
+	for i := 0; i < numberOfIterationsToDial; i++ {
+		err = sc.execute(publicIP, []string{
+			"n=0; while [ \"$n\" -lt 10 ] && [ ! -f /var/lib/cloud/instance/boot-finished ]; do echo 'Waiting for cloud-init...'; n=$(( n + 1 )); sleep 1; done",
+			"wget -nv -O ravendb.deb " + ravenPackageUrl,
+			"sudo apt-get update -y",
+			"sudo apt-get install -y -f ./ravendb.deb",
+		}, stdoutBuf, conn)
+		if err == nil {
+			break
+		} else {
+			if i != numberOfIterationsToDial {
+				time.Sleep(time.Second)
+			} else {
+				return err
+			}
+		}
+	}
+
+	err = upload(conn, stdoutBuf, "/etc/ravendb/license.json", sc.License)
+	if err != nil {
+		return err
+	}
+
+	contents, err := readFileContents("/etc/ravendb/settings.json", stdoutBuf, conn)
+	if err != nil {
+		return err
+	}
+
+	var settings map[string]interface{}
+	err = json.Unmarshal(contents, &settings)
+	if err != nil {
+		stdoutBuf.WriteString("Failed to parse JSON\n")
+		stdoutBuf.Write(contents)
+		return err
+	}
+
+	if sc.ClusterCertificate != nil && sc.Insecure == false {
+		settings["Security.Certificate.Path"] = "/etc/ravendb/certificate.pfx"
+		err = upload(conn, stdoutBuf, "/etc/ravendb/certificate.pfx", sc.ClusterCertificate)
+		if err != nil {
+			return err
+		}
+	}
+
+	scheme := "https"
+	if sc.Insecure {
+		settings["Security.UnsecuredAccessAllowed"] = "PublicNetwork"
+		scheme = "http"
+	}
+	httpUrl, err := sc.setupUrls(index, scheme, settings)
+	if err != nil {
+		return err
+	}
+
+	settings["ServerUrl"] = scheme + "://0.0.0.0:" + strconv.Itoa(sc.Url.HttpPort)
+	settings["ServerUrl.Tcp"] = "tcp://0.0.0.0:" + strconv.Itoa(sc.Url.TcpPort)
+	settings["Setup.Mode"] = "None"
+	settings["License.Path"] = "/etc/ravendb/license.json"
+
+	for key, value := range sc.Settings {
+		settings[key] = value
+	}
+
+	jsonOut, err := json.MarshalIndent(settings, "", "\t")
+	if err != nil {
+		return err
+	}
+
+	err = upload(conn, stdoutBuf, "/etc/ravendb/settings.json", jsonOut)
+	if err != nil {
+		return err
+	}
+	err = sc.execute(publicIP, []string{
+		"sudo chown -R ravendb:ravendb /etc/ravendb/",
+		"sudo systemctl restart ravendb",
+		"curl -v --retry-connrefused --retry 100 --retry-delay 1 " + httpUrl + "/setup/alive",
+	}, stdoutBuf, conn)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (sc *ServerConfig) setupUrls(index int, scheme string, settings map[string]interface{}) (string, error) {
+
+	httpUrl, tcpUrl, err := sc.GetUrlByIndex(index, scheme)
+	if err != nil {
+		return "", err
+	}
+	settings["PublicServerUrl"] = httpUrl
+	settings["PublicServerUrl.Tcp"] = tcpUrl
+	return httpUrl, nil
+}
+
+func (sc *ServerConfig) GetUrlByIndex(index int, scheme string) (string, string, error) {
+
+	if sc.Url.HttpPort == 0 {
+		if sc.Insecure == false {
+			sc.Url.HttpPort = 443
```

Comment:
443 and 8080 and other port numbers show up again. Can you make it a constant somewhere and use it? E.g. `DEFAULT_SECURE_RAVENDB_HTTP_PORT`